media/fxp.ksy

```diff
@@ -0,0 +1,80 @@
+meta:
+  id: fxp
+  title: VST 2.x plugin preset file
+  file-extension: fxp
+  license: CC0-1.0
```

Comment:
It is likely it cannot be CC0 if it is derived from proprietary header files. Not your attorney, though.

Reply:
I am skeptical about that too, what else can it be? The SDK itself is distributed no more, but VST2 plugins are common

Reply:
I have just googled and found https://github.com/Xaymar/vst2sdk

Reply:
The legality of that is questionable, frankly I don't care, but it doesn't include the contents of vstfxstore.h

Reply:
Then probably there is no other legal way as black box or clean room reverse engineering of the format. Black box means just looking on multiple hex dumps of files. Diffing (sequence alignment) of similar files is also extremily useful. But you cannot do that after you have written the spec using the info from headers. Someone else has to do it.

Reply:
Uhh, this sucks. Maybe I can submit a spec for VST3 plugins, that ones under GPLv3

Reply:
>Uhh, this sucks

Yes, it really sucks to waste time and manpower on reversing something that is "source available". But the your spec doesn't look very complex, fully black box reversing probably can be done by a person knowing what is supposed to be in the file and having files converted from it.

Reply:
>But, honestly no one is gonna come here claiming copyright either

You underestimate copyright trolls. They have come for re3 (an alternative open-source engine for GTA games, and now there is litigation). They have come for youtube-dl once. It's only beginning.

Reply:
@generalmimon, I don't see anything funny in the situation. 

https://storage.courtlistener.com/recap/gov.uscourts.cand.384429/gov.uscourts.cand.384429.1.0.pdf

Reply:
> derived from proprietary header files

Header files rarely contain copyrightable information. Even the strictest copyright defenders agree. For example see https://softwareengineering.stackexchange.com/a/216480.

Can you provide a single person or reference claiming that purely looking at a header file and using 

> Not your attorney

Thank godness, with this mindset "stricter than the Pope" of yours you would lose every case in favor of the plaintiff. Life is not black and white, regardless of whether you say it is. We don't work in binary as computers, we live in a society. As you like to say, the border between fair use and copyright infringement is fuzzy, but if you knew what copyright is and what it protects (which you apparently don't know, or at least you're unable to use these facts in practice, or just your unfounded anxiety about potential copyright infringement is stronger than common sense, I don't which it is), you would know that you can for example:

1. read a recipe and cook a meal accordingly (without needing to print a license of the recipe book and attach it to each dish before serving, and stick a label to the plate with copyright notice of the recipe), or
2. learn principles from a math textbook and afterwards write a math proof about some theorem that doesn't use any sentence from the textbook, but you may use some known theorems and probably notation learnt from that (will you publish a book with that proof you wrote with declaring textbook authors as copyright holders and the license of the textbook as the license of your proof?),
3. read a Wikipedia article, let it settle for a few days (to make sure you remember the message, not concrete phrases) and then write a blog post about what you have learnt from that (your text will certainly have entirely different structure, because you don't remember what the sentences on Wikipedia were and how the text was broken down into paragraphs or sections, and you will also not remember any long phrases from there, so you won't be closely paraphrasing or anything like that) - then your blog post is original (uses only uncopyrightable facts and perhaps some scientific terms, which also cannot be copyrighted).

How are these examples any different from learning information about some file format structure and then creating a format specification describing the same structures?

Perhaps you read somewhere that _structure_ may also be copyright protected, and that's why you think you always create a derivative work from where you find out how a file format struct is laid out. But that _structure_ which may be subject to copyright means the structure of the _text_, or a document, i.e. what sections are there, how many paragraphs, whether it uses floating pictures on the right/left, what phrasing the document uses in each sentence, what order the sentences are presented in, how the text is formatted and so on. It has nothing to do with the information presented there. If it only presents factual data which are simply true no matter what (but it may be presented with fancy formatting and whatnot), you can use the facts presented there, write them in your own words, forget about how the original document was formatted and format it yourself in a way you can think of. Then the facts you present there are not yours, the facts are still freely usable, but your document (your phrasing, document structure and formatting) is truly yours and you have no obligation to credit anyone (but yourself, if you don't release the work in public domain, which you can too).

On the other hand, "file format structure" is an entirely different thing - since there is a consensus that a file format itself cannot be copyrighted, information about output data structure is basically just facts.

> @generalmimon, I don't see anything funny in the situation.

Don't you? To me it's comical (and sad at the same time) at this point that you still believe you can't look at a `struct` definition in a C/C++ header file and use the facts about what order the fields are written in your implementation, without having to give all credits to the header file author (who likely also has information about the structure from another source, but is knowledgeable enough to know that the binary structure layout is something that cannot be copyrighted) and declaring a license of the project (where the header file is coming from) on your file. I have no idea how you came up with this and where you read or heard it, and I'm really curious to know.

When I see your specifications like this one (https://github.com/kaitai-io/kaitai_struct_formats/pull/392/files#diff-46fdcf71a1c1c12db7cb8a44d5e3474e86e7147cd1b2787da4ef8ba1b58fe11eR7) where you declare `GPL-2.0-or-later` as the license literally **just** because you looked at **this exact table** from https://www.gpsbabel.org/htmldoc-development/fmt_gnav_trl.html:

> Position | Data type | Field info
> -- | -- | --
> 0 | 32-bit signed int | Unix timestamp
> 4 | 32-bit float | Latitude
> 8 | 32-bit float | Longitude
> 12 | 32-bit float | Altitude (!rotated left by eight bits!)

... (seriously, it's only this table! - I'm not making this shit up), I'd like to know why you did this. You are not creating a derivative work by using these facts in your specification, so why you are declaring the same license regardless? From this perspective, it looks like putting `GPL-2.0-or-later` must be your willful decision, because there is no other logical reason why you would do that, but in that case I would argue that it's pointless to license that trivial file under GPL, because anyone can just look at the structure defined by your specification and reimplement it anywhere else without caring about copyright at all, exactly because the field layout is not copyrightable no matter where it is presented.

Reply:
>Header files rarely contain copyrightable information (i.e. if you don't count long comments, which can contain arbitrary text). Even the strictest copyright defenders agree. For example see https://softwareengineering.stackexchange.com/a/216480.

No, they do contain copyright information. Anything that is a product of creativity is copyrightable.


>Thank godness, with this mindset "stricter than the Pope"

I don't have such a mindset. I don't consider copyright as good. I don't consider people using copyright as good.

>of yours you would lose every case in favor of the plaintiff.

Everyone will. It depends not in mindset but on pkwer distribution and power relations.

>Life is not black and white, regardless of whether you say it is. We don't work in binary as computers, we live in a society.

This kind of society is the worst thing we see.

>As you like to say, the border between fair use and copyright infringement is fuzzy

All laws are "fuzzy" and the way they are applied depends more on who is plaintiff, who is the defendant, who is the jury, who are the attorneys, how all of them are related to each other, how all of them are related to other powerful entities.

>but if you knew what copyright is and what it protects

Copyright, as all the other laws, protects the interests of powerful entities. It is the only thing any law protects. Everyone who have benefited from any law got it only as a side effect.

>recipe example

clearly can be a violation.

>Wikipedia article example

probably not. One can always sue you wor that, but you can always claim you have never read the Wikipedia article, if you have accessed it via Tor.

>math textbook theorems example

Theorems are explicitly out of copyright. But proofs can be colyrighted because they are results of creativity.

>How are these examples any different from learning information about some file format structure and then creating a format specification describing the same structures?

* File format structures are the things companies benefit for their knowledge being their monopoly;

* The companies are taxpayers. Less their revenue - less the taxes.

* The companies belong to powerful people with ties to elites.

> Perhaps you read somewhere that structure may also be copyright protected, and that's why you think you always create a derivative work from where you find out how a file format struct is laid out.

No, it is file format structure can be [protected](https://en.wikipedia.org/wiki/Protection_racket) as a result of human creativity.

>But that structure which may be subject to copyright means the structure of the text, or a document, i.e. what sections are there, how many paragraphs, whether it uses floating pictures on the right/left, what phrasing the document uses in each sentence, what order the sentences are presented in, how the text is formatted and so on. It has nothing to do with the information presented there.

Completely true.

>If it only presents factual data which are simply true no matter what (but it may be presented with fancy formatting and whatnot)

then a program implementing an algorithm is also "factual data".

>, you can use the facts presented there, write them in your own words, forget about how the original document was formatted and format it yourself in a way you can think of. Then the facts you present there are not yours, the facts are still freely usable, but your document (your phrasing, document structure and formatting) is truly yours and you have no obligation to credit anyone (but yourself, if you don't release the work in public domain, which you can too).

We are not about formatting.

>On the other hand, "file format structure" is an entirely different thing - since there is a consensus that a file format itself cannot be copyrighted, information about output data structure is basically just facts.

It is to be decided by judges on case-by-case basis.


>Don't you? To me it's comical (and sad at the same time) at this point that you still believe you can't look at a struct definition in a C/C++ header file and use the facts about what order the fields are written in your implementation, without having to give all credits to the header file author

Giving all the credits to the original author covers my ass.

>(who likely also has information about the structure from another source, but is knowledgeable enough to know that the binary structure layout is something that cannot be copyrighted)

It can be copyrighted, but I assumme he is a [law-abiding citizen](https://en.wikipedia.org/wiki/Law_Abiding_Citizen) who either got the permission (probably non-public) from the copyright holders or have reversed the impl in a legally-clean way.

>and declaring a license of the project (where the header file is coming from) on your file.

Because without that project there would have been no ksy file. It is explicitly derived from the code of the original project.

>When I see your specifications like this one (https://github.com/kaitai-io/kaitai_struct_formats/pull/392/files#diff-46fdcf71a1c1c12db7cb8a44d5e3474e86e7147cd1b2787da4ef8ba1b58fe11eR7) where you declare GPL-2.0-or-later as the license literally just because you looked at this exact table from https://www.gpsbabel.org/htmldoc-development/fmt_gnav_trl.html:

>... (seriously, it's only this table! - I'm not making this shit up), I'd like to know why you did this.

That table was created by people of that project. IDK how it was created, but probably in a legal way. There is only 1 thing clear, without the table there would have been no ksy.

>You are not creating a derivative work by using these facts in your specification, so why you are declaring the same license regardless?

It is not the uncopyrightable fact, it is a copyrightable one.

>From this perspective, it looks like putting GPL-2.0-or-later must be your willful decision, because there is no other logical reason why you would do that

The file was clearly derived from their work.

>but in that case I would argue that it's pointless to license that trivial file under GPL, because anyone can just look at the structure defined by your specification and reimplement it anywhere else without caring about copyright at all

Completely OK, but my ass is covered. It is their decision to do such a thing. They are liable for it, not me.